src/main/java/org/springframework/guice/annotation/ModuleRegistryConfiguration.java

```diff
@@ -118,8 +120,12 @@ private void mapBindings(Map<Key<?>, Binding<?>> bindings,
 			ConstructorArgumentValues args = new ConstructorArgumentValues();
 			args.addIndexedArgumentValue(0, key.getTypeLiteral().getRawType());
 			args.addIndexedArgumentValue(1, key);
+			args.addIndexedArgumentValue(2, Scopes.isSingleton(binding));
 			bean.setConstructorArgumentValues(args);
 			bean.setTargetType(ResolvableType.forType(key.getTypeLiteral().getType()));
+			if(!Scopes.isSingleton(binding)) {
+				bean.setScope(ConfigurableBeanFactory.SCOPE_PROTOTYPE);
```

Comment:
For completeness shouldn't we leverage Spring Scope interface and create an adapter from the Guice custom scope?

Reply:
The only pre-defined scopes in both frameworks are Singleton and Prototype. I can't think of any way to better handle any of the user defined scopes in either framework.